### PR TITLE
Create a mixin for list-accumulator

### PR DESCRIPTION
**Pull request by ciss1995**
Context:
US120871 - https://rally1.rallydev.com/#/357252966636d/custom/367300408400?detail=%2Fuserstory%2F438086221156
Create a mixin for list-accumulator

Quality:
Tested the demo page and in Learning Paths

### CODE REVIEW

**Review by AKobets — APPROVED** [review]

File: list-item-accumulator-mixin.js [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+		this.addEventListener('d2l-tooltip-show', () => this._tooltipShowing = true);
+		this.addEventListener('d2l-tooltip-hide', () => this._tooltipShowing = false);
+		this._getActions();
+		super.firstUpdated(changedProperties);
+	}
+	get isOnlyChild() {
+		const nodes = this.parentNode.querySelectorAll('d2l-labs-list-item-accumulator');
+		return nodes.length === 1 ? nodes[0] === this : false;
+	}
+	_renderListItem(illustration, main, secondary, supportingInfo) {
```
Comment (m6jones) [inline list-item-accumulator-mixin.js:229]: Can this be an object parameter list?

Comment (m6jones) [reply list-item-accumulator-mixin.js:229]: Can main be called title.

File: list-item-accumulator-mixin.js [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+							<slot name="illustration" class="d2l-activitiy-collection-list-item-illustration">${illustration}</slot>
+							<div class="d2l-list-item-main">
+								<slot class="d2l-body-standard">${main}</slot>
+								<slot class="d2l-body-small" name="secondary">${secondary}</slot>
+								<slot class="d2l-body-compact" name="supporting-info">${supportingInfo}</slot>
+							</div>
+						</div>
+						<div slot="actions">
+							<div class="d2l-list-item-actions-container">
+								<slot name="primary-action"></slot>
```
Comment (m6jones) [inline list-item-accumulator-mixin.js:265]: These will need to have the defaults as well?

File: list-item-accumulator-mixin.js [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+								<d2l-dropdown-more text="${this.localize('actions')}" class="${classMap(dropdownClasses)}">
+									<d2l-dropdown-menu id="${this._dropdownId}">
+										<d2l-menu label="${this.localize('secondaryActions')}">
+											${mobilePrimaryAction}
+											${this._renderReorderActions()}
+											<slot name="secondary-action"></slot>
```
Comment (m6jones) [inline list-item-accumulator-mixin.js:271]: Same as primary-action

File: list-item-accumulator-mixin.js (hunk 1 above)
Comment (ciss1995) [reply list-item-accumulator-mixin.js:229]: Yeah sure!

Comment (m6jones) [reply list-item-accumulator-mixin.js:229]: `_renderListItem(illustration, main, secondary, supportingInfo)` => `_renderListItem({illustration, title, secondary, supportingInfo} = {})`

**Review by m6jones — APPROVED** [review]

**Pull request merged**